qutebrowser/browser/browsertab.py

```diff
@@ -165,12 +164,12 @@ class AbstractAction:
     """Attribute ``action`` of AbstractTab for Qt WebActions."""
 
     # The class actions are defined on (QWeb{Engine,}Page)
-    action_class = None  # type: type
+    action_class: Any = None
     # The type of the actions (QWeb{Engine,}Page.WebAction)
-    action_base = None  # type: type
+    action_base: Any = None
```

Comment:
Thanks for the detailed response. It was deliberate, but it was because I was mistaken in thinking `type` and `Any` were (roughly?) equivalent.

The reason I changed it to `Any` was that `mypy` complained about using `action_base: type = None`. To be precise it complained that I couldn't assign `None` to the variable. If I changed the type to `Optional[type]` then it complained about the `isinstance()` method in `run_string()`. I made an attempt to use a more precise typing as you specified in your comment and was able to do so successfully with `action_class` (although I needed to wrap the type in an `Optional` compared to what you had), but `mypy` still complained about the same `isinstance()` method for `action_base`.

So, it seems that `mypy` behaves differently depending on whether the typing is in annotations or in comments. It seems the comment style doesn't require an `Optional` in the same way, and including the `Optional` somehow causes problems when dealing with the `isinstance()` method.

Looking at this again, the only way around this that I've found (assuming we want to use the annotation style) is to define the types _without_ initialising the variables. This is arguably undesirable as there is no enforcement of `action_class` and `action_base` being set by child classes, and referencing a non-existent member variable will raise an `AttributeError` which I assume is not handled. However even with the current code if the variables are initialised to `None`, then the `isinstance()` call will fail with a `TypeError`. Given that the code already requires that child classes are "well behaved", I've pushed the code change to define the types without initialisation. Happy to hear if you have different opinions of course.

> Also FWIW, if this was a deliberate change, it'd be good if you could "mark" it with a quick comment

You are correct. As I said, I mistakenly thought `type` and `Any` were roughly equivalent, but given that I spent some time already investigating this, and it was an intentional change, I should have called it out. I'll try to do that in the future.

Reply:
Thanks for the quick update! Seems fine to me - like you say, the old code with `None` also relied on subclasses being well-behaved, so the new way seems fine as well.